index.js

```diff
@@ -156,7 +156,22 @@ Archive.prototype.download = function (i, cb) {
     if (err) return cb(err)
 
     var feed = self._getFeed(entry)
-    if (!feed) return cb(null)
+    var dest = join(self.directory, entry.name)
+
+    if (!feed) return createEmtpyEntry()
+
+    function createEmtpyEntry () {
+      var dir = dest
+      if (entry.type === 'file') dir = path.dirname(dest)
+      mkdirp(dir, function (err) {
+        if (err) return cb(err)
+        if (entry.type !== 'file') return cb(null)
+        fs.open(dest, 'a', function (err) {
```

Comment:
fs.open calls the callback with a file descriptor as well. you should close that afterwards

```
fs.open(..., function (err, fd) {
  if (err) return cb(err)
  fs.close(fd, cb)
})
```